MathLib/LinAlg/Eigen/EigenMatrix.h

```diff
@@ -184,6 +185,24 @@ void EigenMatrix::add(std::vector<IndexType> const& row_pos,
     }
 };
 
+/// Sets the sparsity pattern of the underlying EigenMatrix.
+template <typename SPARSITY_PATTERN>
+struct SetMatrixSparsity<EigenMatrix, SPARSITY_PATTERN>
```

Comment:
You are partially specializing a class template from a different header file.
Shouldn't there be a reference to the overridden full class template here?